tests/dataloaders/test_custom_dataloader.py

```diff
@@ -0,0 +1,400 @@
+        early_stopping=False,
+    )
+
+    user_attributes_model_census = model_census._get_user_attributes()
+    # # TODO: do we need to put inside
+    # user_attributes_model_census = \
+    #     {a[0]: a[1] for a in user_attributes_model_census if a[0][-1] == "_"}
+    pprint(user_attributes_model_census)
+    # dataloader_census = model_census._make_data_loader(datamodule.datapipe)
+    # # this casus errors
+    # _ = model_census.get_elbo(dataloader=dataloader_census)
+    # _ = model_census.get_marginal_ll(dataloader=dataloader_census)
+    # _ = model_census.get_reconstruction_error(dataloader=dataloader_census)
+    # _ = model_census.get_latent_representation(dataloader=dataloader_census)
+
+    model_census.save(save_path, overwrite=True)
+    model_census2 = scvi.model.SCVI.load(save_path, adata=False)
+
+    model_census2.train(
+        datamodule=datamodule,
+        max_epochs=max_epochs,
+        batch_size=batch_size,
+        train_size=train_size,
+        early_stopping=False,
+    )
+
+    user_attributes_model_census2 = model_census2._get_user_attributes()
+    pprint(user_attributes_model_census2)
+    # dataloader_census2 = model_census2._make_data_loader()
+    # this casus errors
+    # _ = model_census2.get_elbo()
+    # _ = model_census2.get_marginal_ll()
+    # _ = model_census2.get_reconstruction_error()
+    # _ = model_census2.get_latent_representation()
+
+    # takes time
+    adata = cellxgene_census.get_anndata(
+        census,
+        organism=experiment_name,
+        obs_value_filter=obs_value_filter,
+        var_coords=hv_idx,
+    )
+
+    # TODO: do we need to put inside (or is it alrady pre-made) - perhaps need to tell CZI
+    adata.obs["batch"] = adata.obs[batch_keys].agg("".join, axis=1).astype("category")
+
+    scvi.model.SCVI.prepare_query_anndata(adata, save_path)
+    scvi.model.SCVI.load_query_data(registry=datamodule.registry, reference_model=save_path)
+
+    scvi.model.SCVI.prepare_query_anndata(adata, model_census2)
+
+    scvi.model.SCVI.setup_anndata(adata, batch_key="batch")  # needed?
+    model_census3 = scvi.model.SCVI.load(save_path, adata=adata)
+
+    model_census3.train(
+        datamodule=datamodule,
+        max_epochs=max_epochs,
+        batch_size=batch_size,
+        train_size=train_size,
+        early_stopping=False,
+    )
+
+    user_attributes_model_census3 = model_census3._get_user_attributes()
+    pprint(user_attributes_model_census3)
+    _ = model_census3.get_elbo()
+    _ = model_census3.get_marginal_ll()
+    _ = model_census3.get_reconstruction_error()
+    _ = model_census3.get_latent_representation()
+
+    scvi.model.SCVI.prepare_query_anndata(adata, save_path, return_reference_var_names=True)
+    scvi.model.SCVI.load_query_data(adata, save_path)
+
+    scvi.model.SCVI.prepare_query_anndata(adata, model_census3)
+    scvi.model.SCVI.load_query_data(adata, model_census3)
+
+    datamodule_inference = CensusSCVIDataModule(
```

Comment:
check here that using different genes and different batches fails. You can take much fewer cells here, like 1000.